lib/samples/navigate_route_with_rerouting/navigate_route_with_rerouting.dart

```diff
@@ -0,0 +1,563 @@
+                      Expanded(
+                        child: Text(
+                          'Route status: $_routeStatus',
+                          style: Theme.of(context).textTheme.labelMedium,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(10, 0, 8, 8),
+                  child: Row(
+                    children: [
+                      Expanded(
+                        child: Text(
+                          'Distance remaining: $_remainingDistance',
+                          style: Theme.of(context).textTheme.labelMedium,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(10, 0, 8, 8),
+                  child: Row(
+                    children: [
+                      Expanded(
+                        child: Text(
+                          'Time remaining: $_remainingTime',
+                          style: Theme.of(context).textTheme.labelMedium,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(10, 0, 8, 24),
+                  child: Row(
+                    children: [
+                      Expanded(
+                        child: Text(
+                          'Next direction: $_nextDirection',
+                          style: Theme.of(context).textTheme.labelMedium,
+                          softWrap: true,
+                          maxLines: 2,
+                          overflow: TextOverflow.ellipsis,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+            // Display a progress indicator.
+            Visibility(
+              visible: _initNavigation,
+              child: const Center(
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    LinearProgressIndicator(),
+                  ],
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    _flutterTts.stop();
+    _simulatedLocationDataSource?.stop();
+
+    super.dispose();
+  }
+
+  // Set up the map with a navigation basemap style.
+  Future<void> onMapViewReady() async {
+    // Create a map with a navigation basemap style.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISNavigation);
+    _mapViewController.arcGISMap = map;
+
+    map.onLoadStatusChanged.listen(
```

Comment:
I'm not sure if it's important for the subscription returned by `listen` to be captured by a variable so it can be canceled in the `dispose()`. But that can be avoided if this was written differently:
```dart
map.load().then((_) async {
  setState(() => _initNavigation = true);
  await initRouteTask();
  setState(() => _initNavigation = false);
});
```
I think that's over all simpler.